openupgrade_scripts/scripts/account/15.0.1.2/upgrade_analysis_work.txt

```diff
@@ -0,0 +1,199 @@
+account      / account.reconcile.model.template / match_total_amount_param (float): DEL
+account      / account.reconcile.model.template / payment_tolerance_param (float): NEW hasdefault
+account      / account.reconcile.model.template / payment_tolerance_type (selection): NEW required, selection_keys: ['fixed_amount', 'percentage'], req_default: function, hasdefault
+# DONE: fill value for payment_tolerance_type
+
+account      / account.tax              / country_id (many2one)         : NEW relation: res.country, required
+account      / account.tax              / is_base_affected (boolean)    : NEW hasdefault
+# DONE: fill value for country_id will be handled in module l10n_*
+
+account      / account.tax.carryover.line / amount (float)                : NEW required, req_default: function, hasdefault
+account      / account.tax.carryover.line / company_id (many2one)         : NEW relation: res.company, required, req_default: function, hasdefault
+account      / account.tax.carryover.line / date (date)                   : NEW required, req_default: function, hasdefault
+account      / account.tax.carryover.line / foreign_vat_fiscal_position_id (many2one): NEW relation: account.fiscal.position
+account      / account.tax.carryover.line / name (char)                   : NEW required
+account      / account.tax.carryover.line / tax_report_line_id (many2one) : NEW relation: account.tax.report.line
+# NOTHING TO DO: new model
+
+account      / account.tax.group        / country_id (many2one)         : NEW relation: res.country
+account      / account.tax.group        / preceding_subtotal (char)     : NEW
+# NOTHING TO DO: new feature
+
+account      / account.tax.report.line  / carry_over_condition_method (selection): NEW selection_keys: ['always_carry_over_and_set_to_0', 'no_negative_amount_carry_over_condition']
+account      / account.tax.report.line  / carry_over_destination_line_id (many2one): NEW relation: account.tax.report.line
+account      / account.tax.report.line  / carryover_line_ids (one2many) : NEW relation: account.tax.carryover.line
+account      / account.tax.report.line  / is_carryover_persistent (boolean): NEW hasdefault
+account      / account.tax.report.line  / is_carryover_used_in_balance (boolean): NEW
+# NOTHING TO DO: new feature
+
+account      / account.tax.template     / is_base_affected (boolean)    : NEW hasdefault
+# NOTHING TO DO: new feature
+
+account      / product.template         / account_tag_ids (many2many)   : NEW relation: account.account.tag
+# NOTHING TO DO: new feature
+
+account      / res.company              / account_fiscal_country_id (many2one): NEW relation: res.country, isfunction: function, stored
+account      / res.company              / account_journal_payment_credit_account_id (many2one): NEW relation: account.account
+account      / res.company              / account_journal_payment_debit_account_id (many2one): NEW relation: account.account
```

Comment:
@tranngocson1996 đối với những công ty đã có COA, cần tìm tài khoản `Outstanding Payments / Outstanding Receipts` để điền cho 2 trường này, nếu chưa có thì cần tạo mới nhé

Reply:
@royleviindoo em nghĩ chỗ này không cần thiết anh ạ. Vì
1.Hai trường này không required người dùng có thể vào chọn taì khoản sau nếu muốn
2.Các tài khoản `Outstanding Payments / Outstanding Receipts` trong từng sổ nhật Ký đã được em cập nhật vào trong từng `account_payment_method_line` của từng sổ rồi anh. Nên vẫn sẽ hoạt đông bth như 14 khi có phát sinh các tài khoản vẫn như cũ (vd thanh toán)

Reply:
@tranngocson1996 Nên 15.0 luồng đang if else theo `company_id.account_journal_payment_debit_account_id`, nên cần phải điền nhé
Giả xử dữ liệu mới, khi load COA mới, thì có sinh ra tài khoản để điền vào, nên dữ liệu cũ cũng phải như vậy nhé

Reply:
@royleviindoo dạ ok anh ạ. Vậy thì để em điền vào 2 tài khoản đó trong công ty